tests/core/actions/test_action_run_slot_rejections.py

```diff
@@ -0,0 +1,421 @@
+                    "utter": None,
+                }
+            ],
+            "type": "pattern_collect_information",
+        },
+    ]
+
+    tracker = DialogueStateTracker.from_events(
+        sender_id=uuid.uuid4().hex,
+        evts=[
+            UserUttered("i want to setup a new recurrent payment."),
+            SlotSet("recurrent_payment_type", "international transfer"),
+            SlotSet("dialogue_stack", dialogue_stack),
+        ],
+        slots=[
+            TextSlot("recurrent_payment_type", mappings=[]),
+            AnySlot("dialogue_stack", mappings=[]),
+        ],
+    )
+
+    action_run_slot_rejections = ActionRunSlotRejections()
+    events = await action_run_slot_rejections.run(
+        output_channel=default_channel,
+        nlg=rejection_test_nlg,
+        tracker=tracker,
+        domain=rejection_test_domain,
+    )
+
+    assert events == [SlotSet("recurrent_payment_type", None)]
+
+    out = capsys.readouterr().out
+    assert "[error    ] collect.rejection.missing.utter" in out
+    assert "utterance=None" in out
+
+
+async def test_action_run_slot_rejections_top_frame_slot_not_found_utter(
+    default_channel: OutputChannel,
+    rejection_test_nlg: TemplatedNaturalLanguageGenerator,
+    rejection_test_domain: Domain,
+    capsys: CaptureFixture,
+) -> None:
+    dialogue_stack = [
+        {
+            "frame_id": "4YL3KDBR",
+            "flow_id": "setup_recurrent_payment",
+            "step_id": "ask_payment_type",
+            "frame_type": "regular",
+            "type": "flow",
+        },
+        {
+            "frame_id": "6Z7PSTRM",
+            "flow_id": "pattern_ask_collect_information",
+            "step_id": "start",
+            "collect_information": "recurrent_payment_type",
+            "rejections": [
+                {
+                    "if": 'not ({"direct debit" "standing order"} contains recurrent_payment_type)',  # noqa: E501
+                    "utter": "utter_not_found",
+                }
+            ],
+            "type": "pattern_collect_information",
+        },
+    ]
+
+    tracker = DialogueStateTracker.from_events(
+        sender_id=uuid.uuid4().hex,
+        evts=[
+            UserUttered("i want to setup a new recurrent payment."),
+            SlotSet("recurrent_payment_type", "international transfer"),
+            SlotSet("dialogue_stack", dialogue_stack),
+        ],
+        slots=[
+            TextSlot("recurrent_payment_type", mappings=[]),
+            AnySlot("dialogue_stack", mappings=[]),
+        ],
+    )
+
+    action_run_slot_rejections = ActionRunSlotRejections()
+    events = await action_run_slot_rejections.run(
+        output_channel=default_channel,
+        nlg=rejection_test_nlg,
+        tracker=tracker,
+        domain=rejection_test_domain,
+    )
+
+    assert events == [SlotSet("recurrent_payment_type", None)]
+
+    out = capsys.readouterr().out
+    assert "[debug    ] collect.rejection.failed.finding.utter" in out
```

Comment:
this would need to be changed if you change the case to an error log in the action

Reply:
Done: https://github.com/RasaHQ/rasa/pull/12802/commits/79b1cd1224fa7df417000941b36c3619a8e12ee9